### PR TITLE
Patch for blog comments embedding

### PR DESCRIPTION
**Pull request by BronyBorn**
This was a weird bug that took me a while to figure out that I had partially fixed it before.
Apparently the embeds for comments (specifically the comments APP) was being sent the proper identifier value only part of the time. After scrubbing the code I noticed that ```index.php``` used the same method as ```postfeed.php```, of which got fixed a few months ago. So I fixed that issue as well as removing the comments count link when the ```$post->full == true``` since that links to the same page (plus the count value doesn't auto update so it looks broken, lol).

I also removed the post title from the identifier URL. The reason this was done was because of the fact that the title is a non-essential part of the post's URL to the blog, but WAS for the comment embeds. So different comments would show up for the same post depending on whether the title was in the URL or not. This has been fixed.

(Feel free to ignore the extended field commits from a month ago, pushed them to the wrong branch lol)

### CODE REVIEW

**Comment by jbroadway** [comment]
Does removing the post title from the identifier URL change the permalink URL anywhere else?

I include the post title in the URL for SEO purposes, and it's a good idea too to make sure there's only one canonical URL for each post to prevent diluting SEO across two URLs for the same page.

**Comment by BronyBorn** [comment]
Did a bit more looking around and fixed a couple permalinks. 
The update to the identifier comes from the issue that arises when someone goes to the url without the title in it or if the title changes so would the identifier prior to this update. All those variations would produce different comment embeds depending on the url instead of a single one for that blog post.

File: apps/blog/views/post.html [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 <div class="blog-post{% if full %} blog-post-full{% end %} block">
 {% if ! full %}
-<h3><a href="{{ url }}">{{ title }}</a></h3>
+<h3><a href="{{ fullurl }}">{{ title }}</a></h3>
+{% else %}
+<h3>{{ title }}</h3>
```
Comment (jbroadway) [inline apps/blog/views/post.html]: Won't this `else` show the post title twice? We set `$page->title` to the post's title already in the `blog/post` handler.

File: apps/blog/handlers/index.php [hunk 2]
```diff
@@ -40,10 +40,12 @@
...
 
-	foreach ($posts as $post) {
-		$post->url = '/blog/post/' . $post->id . '/' . URLify::filter ($post->title);
+	foreach ($posts as $_post) {
+		$post = $_post->orig();
+		$post->url = '/blog/post/' . $post->id . '/';
+		$post->fullurl = $post->url . URLify::filter ($post->title);
 		$post->tag_list = (strlen ($post->tags) > 0) ? explode (',', $post->tags) : array ();
-		$post->social_buttons = Appconf::blog ('Social Buttons');
+		$post->social_buttons = $appconf['Social Buttons'];
```
Comment (jbroadway) [inline apps/blog/handlers/index.php]: I'd like to keep the `Appconf::*` style access to app settings. I'm thinking of eventually deprecating `$appconf` so the settings can be lazy-loaded and not automatically parsed on requests that don't refer to any settings (by removing this: https://github.com/jbroadway/elefant/blob/master/lib/Controller.php#L466-L467).

**Comment by BronyBorn** [comment]
Fixed the points you brought up and updated an incorrect href to the fullurl value.

**Comment by BronyBorn** [comment]
I've been looking around to make sure the URLs are updated and I haven't found any more needing fixed. Can this be merged or is there there anything else in this PR that I should fix up?

**Comment by jbroadway** [comment]
Sorry, I really gotta get on these quicker!

**Pull request merged**